src/filters/oauth-exception-filter.ts

```
@@ -0,0 +1,31 @@
+  Injectable
+} from "@nestjs/common"
+
+@Catch()
+@Injectable()
+export class OAuthExceptionFilter implements ExceptionFilter {
+  private static _provider: "google" | "github"
+
+  constructor(private readonly env: EnvService) {}
+
+  static provider(provider: "google" | "github"): typeof OAuthExceptionFilter {
+    this._provider = provider
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)